frontend/src/pages/dashboardtabs/Tanks.tsx

```diff
@@ -26,17 +25,23 @@ import {
 } from '@mui/material';
 import SCDataGrid from '../../components/SCDataGrid';
 import TankGrid from '../../components/datagrid/TankGrid';
+import { useLocation, useNavigate } from 'react-router-dom';
 
 export function TankTabs({
   tanks,
-  employeeId
+  employeeId,
+  selectedTankId,
+  setSelectedTankId
 }: {
   tanks: UpdateTankMetaData[];
   employeeId: number;
+  selectedTankId: number | null;
+  setSelectedTankId(tankId: number | null): void;
 }) {
-  const [selectedTank, setSelectedTank] = useState<
-    UpdateTankMetaData | undefined
-  >(tanks.at(0));
+  const selectedTank = useMemo(
+    () => tanks.find((tank) => tank.id === selectedTankId) ?? null,
```

Comment:
The memo returns null if no tank is selected from the linking feature, but selecting a client in the tanks tab no longer has a default behavior. Maybe instead of returning null we return the first index of the tanks array if there are tanks to display and handle the case when there are no tanks as well.

Reply:
fixed with a use effect.